File: code.leetcode/lianbiao/e21.合并两个有序链表.py
 
class ListNode(object):
    def __init__(self, val=0, next=None):
       self.val = val
       self.next = next
class Solution(object):
	def mergeTwoLists(self, l1: ListNode, l2: ListNode):
		preHead = ListNode(-1) # 创建一个虚拟头节点
		prev = preHead               # 使用 prev 指针追踪当前应该连接的位置

		while l1 and l2 :
			if l1.Val <= l2.Val :
				prev.next = l1
				l1 = l1.next
			else: 
				prev.next = l2
				l2 = l2.next
		
			prev = prev.next # 移动 prev 指针

		if l1 :
			prev.next = l1
		else: 
			prev.next = l2

		return preHead.next

File: code.leetcode/lianbiao/e21.合并两个有序链表.cpp


 struct ListNode{
      int val;
      ListNode *next;
      ListNode() : val(0), next(nullptr) {}
      ListNode(int x) : val(x), next(nullptr) {}
      ListNode(int x, ListNode *next) : val(x), next(next) {}
 };

class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
    ListNode* preHead =new ListNode(-1); // 创建一个虚拟头节点
	
    ListNode* prev = preHead;               // 使用 prev 指针追踪当前应该连接的位置

		while(l1!=nullptr && l2!=nullptr) {
			if (l1->val <= l2->val) {
				prev->next = l1;
				l1 = l1->next;}
			else{ 
				prev->next = l2;
				l2 = l2->next;}
		
			prev = prev->next;
			} // 移动 prev 指针

		if (l1!=nullptr) {
			prev->next = l1;}
		else{ 
			prev->next = l2;}

		return preHead->next ;   
    }
};